ktor-features/ktor-webjars/src/io/ktor/webjars/Webjars.kt

```diff
@@ -0,0 +1,85 @@
+package io.ktor.webjars
+
+import io.ktor.application.ApplicationCall
+import io.ktor.application.ApplicationCallPipeline
+import io.ktor.application.ApplicationFeature
+import io.ktor.application.call
+import io.ktor.http.ContentType
+import io.ktor.http.HttpMethod
+import io.ktor.http.HttpStatusCode
+import io.ktor.http.defaultForFilePath
+import io.ktor.pipeline.PipelineContext
+import io.ktor.request.httpMethod
+import io.ktor.request.uri
+import io.ktor.response.respond
+import io.ktor.response.respondBytes
+import io.ktor.util.AttributeKey
+import org.webjars.MultipleMatchesException
+import org.webjars.WebJarAssetLocator
+import java.nio.file.Paths
+
+class Webjars(val configuration: Configuration) {
+
+    private fun fileName(path: String) : String = Paths.get(path).fileName.toString()
+
+    private fun extractWebJar(path : String) : String {
+        val firstSlash = if (path.startsWith("/")) 1 else 0
+        val nextSlash = path.indexOf("/", 1)
+        val webjar = if(nextSlash > -1) path.substring(firstSlash,  nextSlash) else ""
+        val partialPath = path.substring(nextSlash + 1)
+        return locator.getFullPath(webjar, partialPath)
+    }
+
+    private val locator = WebJarAssetLocator()
+
+    class Configuration {
+        var path = "/webjars"
+        set(value) {
+            var buffer = StringBuilder(value)
```

Comment:
Not sure about it. Most likely it's better to normalize it later. It looks strange if a property modifies values that has been written.

Reply:
It can be changed for sure, but I usually leverage setters for this purpose, to make sure that any incoming change conforms to the expected value of the parent class. Not unusual to see on many places this pattern. But if this is not something kotlin or ktor likes (to me is a coding style more than anything) it can indeed be done somewhere else, perhaps on a init{} block to normalize the path